spec/controllers/automated_tests_controller_spec.rb

```diff
@@ -2,7 +2,7 @@
   include AutomatedTestsHelper
 
   # TODO: add 'role is from a different course' shared tests to each route test below
-  let(:assignment) { create :assignment }
+  let(:assignment) { create :assignment_with_criteria_and_test_results }
```

Comment:
Do not change this here; only in the block that you've changed the tests for (note that you can override a `let` with another `let` that defines the same variable).

Also, instead of using an existing factory that generates extraneous data, create a new factory that just contains test groups. In fact you can create two, one for student test groups and one with instructor test groups.

Similarly, you should modify the existing test group factory to initialize the category to `[]` (the default), but then create new factories to initialize it to `['student']`, etc. Then you can use these factories in the definitions, rather than modifying the objects in the test cases.